### PR TITLE
Feature/points feed service

### PR DESCRIPTION
**Pull request by JanuszBartosz**

### CODE REVIEW

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	@SuppressWarnings("unchecked")
+	@RequestMapping(value = "/available_awards", method = RequestMethod.GET)
+	public ModelAndView awardsView(
+			@RequestParam(value = "sortBy", defaultValue = "name", required = false) String sortProperty,
+			@RequestParam(value = "direction", defaultValue = "ASC", required = false) Sort.Direction direction,
+			@RequestParam(value = "category", required = false) Category category) {
+
+		ModelAndView model = new ModelAndView("available_awards");
+		RestTemplate restTemplate = new RestTemplate();
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java:29]: Please change it to bean

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		Map<Category, List<AwardDTO>> map = new HashMap<>();
+
+		if (category != null) {
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java:32]: Please refactor it to use only 1 if statement - when category is not null

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/PointsFeedController.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+@RestController
+public class PointsFeedController {
+
+	@Autowired
+	PointsService pointsService;
+
+	@ResponseBody
+	@RequestMapping(value = "/new_points", method = RequestMethod.POST)
+	public String doPost(@RequestBody NewPointsRequestDTO newPointsRequestDTO) {
+		System.out.println(newPointsRequestDTO);
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/PointsFeedController.java:22]: TBR or user logger library

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+			map = restTemplate.getForObject("http://localhost:9000/awards?category=" + category + "&sortBy="
+					+ sortProperty + "&direction=" + direction, Map.class);
+		}
+		else {
+			map = restTemplate.getForObject(
+					"http://localhost:9000/awards?sortBy=" + sortProperty + "&direction=" + direction, Map.class);
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java:38]: extract address to properties file (parametrized)

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.pgs.soft.domain.Message;
+import com.pgs.soft.repository.CardRepository;
+
+@Service
+public class GeneratorService {
+
+	RestTemplate restTemplate = new RestTemplate();
```
Comment (kfirsztPgs) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:19]: Please change to bean

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	Random rand = new Random();
```
Comment (kfirsztPgs) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:20]: Please use it dorectly in the method - not as a field

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	@Autowired
+	CardRepository cardRepository;
+
+	@Scheduled(fixedRate = 3600000)
+	public void send() {
+
+		List<String> allCardsNumbers = StreamSupport.stream(cardRepository.findAll().spliterator(), false)
+				.map(c -> c.getNumber()).collect(Collectors.toList());
+
+		System.out.println("Sending");
```
Comment (kfirsztPgs) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:31]: TBR = to be removed
Or use logger library

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+		Message message = new Message(allCardsNumbers.get(rand.nextInt(allCardsNumbers.size())),
+				rand.nextInt(1901) + 100);
+		System.out.println(message);
```
Comment (kfirsztPgs) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:34]: TBR r user logger library

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+		String result = restTemplate.postForObject("http://localhost:8080/new_points", message, String.class);
+		System.out.println(result);
```
Comment (kfirsztPgs) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:36]: TBR = to be removed

File: LoyaltyProgram_Intern/src/test/java/com/pgs/soft/service/impl/PointsServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.pgs.soft.domain.Points;
+import com.pgs.soft.dto.NewPointsRequestDTO;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/test/java/com/pgs/soft/service/impl/PointsServiceImplTest.java:12]: Please remove it. It is not used in test.

File: PointsFeedService/src/main/java/com/pgs/soft/ServletInitializer.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
+package com.pgs.soft;
+
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.context.web.SpringBootServletInitializer;
+
+public class ServletInitializer extends SpringBootServletInitializer {
```
Comment (kfirsztPgs) [inline PointsFeedService/src/main/java/com/pgs/soft/ServletInitializer.java:6]: What for is that class needed?

File: PointsFeedService/src/main/java/com/pgs/soft/domain/Card.java [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+public class Card {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer id;
+
+	private String number;
+
+	@Column(name = "user_id")
+	private Integer user;
```
Comment (kfirsztPgs) [inline PointsFeedService/src/main/java/com/pgs/soft/domain/Card.java:21]: not used field

File: PointsFeedService/src/main/java/com/pgs/soft/domain/Message.java [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.Date;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Message {
+
+	private String cardNumber;
+	private Integer points;
+	private String date;
```
Comment (kfirsztPgs) [inline PointsFeedService/src/main/java/com/pgs/soft/domain/Message.java:13]: Date should be send as a LocalDate type.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java (hunk 1 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java:29]: Done

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java (hunk 2 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java:32]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/PointsFeedController.java (hunk 3 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/PointsFeedController.java:22]: removed

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java (hunk 5 above)
Comment (adriankulinski) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:19]: Done

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java (hunk 6 above)
Comment (adriankulinski) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:20]: done

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java (hunk 7 above)
Comment (adriankulinski) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:31]: removed

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java (hunk 8 above)
Comment (adriankulinski) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:34]: removed

File: PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java (hunk 9 above)
Comment (adriankulinski) [inline PointsFeedService/src/main/java/com/pgs/soft/service/GeneratorService.java:36]: removed

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java (hunk 4 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/AwardsController.java:38]: done.

File: LoyaltyProgram_Intern/src/test/java/com/pgs/soft/service/impl/PointsServiceImplTest.java (hunk 10 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/test/java/com/pgs/soft/service/impl/PointsServiceImplTest.java:12]: DOne.

File: PointsFeedService/src/main/java/com/pgs/soft/ServletInitializer.java (hunk 11 above)
Comment (JanuszBartosz) [inline PointsFeedService/src/main/java/com/pgs/soft/ServletInitializer.java:6]: Done.

File: PointsFeedService/src/main/java/com/pgs/soft/domain/Card.java (hunk 12 above)
Comment (JanuszBartosz) [inline PointsFeedService/src/main/java/com/pgs/soft/domain/Card.java:21]: DOne.

File: PointsFeedService/src/main/java/com/pgs/soft/domain/Message.java (hunk 13 above)
Comment (JanuszBartosz) [inline PointsFeedService/src/main/java/com/pgs/soft/domain/Message.java:13]: Not done.

**Pull request merged**